src/test/java/com/connexta/store/StoreITests.java

```diff
@@ -6,541 +6,388 @@
+            .setBody(getResourceAsString(TRANSFORM_DONE_RESPONSE_FILE))
+            .setHeader("Content-Type", MediaType.APPLICATION_JSON_VALUE);
+
+    mockTransformServer.enqueue(transformRequestResponse);
+    mockTransformServer.enqueue(transformPollDoneResponse);
+
+    final String testFileContent = getResourceAsString(TEST_FILE_PATH);
+    mockTransformServer.enqueue(new MockResponse().setResponseCode(200).setBody(testFileContent));
+
+    // when
+    storeTestClient
+        .post()
+        .uri("/ingest")
+        .contentType(MediaType.MULTIPART_FORM_DATA)
+        .body(BodyInserters.fromValue(TEST_INGEST_REQUEST_BODY))
+        .header(ACCEPT_VERSION, ingestVersion)
+        .header("Last-Modified", LAST_MODIFIED_DATE)
+        .exchange();
+
+    RecordedRequest transformRequest = mockTransformServer.takeRequest();
+    final String datasetId =
+        (String) new JSONObject(transformRequest.getBody().readUtf8()).get("datasetId");
+
+    final String expectedIrmFileName = "irm-" + datasetId + ".xml";
+    final String expectedMetacardFileName = "metacard-" + datasetId + ".xml";
+
+    await()
+        .atMost(10, TimeUnit.SECONDS)
+        .until(
+            () ->
+                dataIsAvailable(
+                    datasetId,
+                    "irm",
+                    APPLICATION_DNI_TDF_XML,
+                    expectedIrmFileName,
+                    testFileContent));
+
+    await()
+        .atMost(10, TimeUnit.SECONDS)
+        .until(
+            () ->
+                dataIsAvailable(
+                    datasetId,
+                    "file",
+                    MediaType.TEXT_PLAIN_VALUE,
+                    ORIGINAL_FILENAME,
+                    TEST_FILE_CONTENT));
+
+    await()
+        .atMost(10, TimeUnit.SECONDS)
+        .until(
+            () ->
+                dataIsAvailable(
+                    datasetId,
+                    "metacard",
+                    MediaType.APPLICATION_XML_VALUE,
+                    expectedMetacardFileName,
+                    TEST_METACARD_CONTENT));
+  }
+
+  private boolean dataIsAvailable(
+      String datasetId,
+      String dataType,
+      String expectedContentType,
+      String expectedFilename,
+      String expectedContent) {
+    try {
+      storeTestClient
+          .get()
+          .uri(
+              UriComponentsBuilder.fromUriString(storeUrl)
+                  .path("/dataset/{datasetId}/{dataType}")
+                  .build(datasetId, dataType)
+                  .toASCIIString())
+          .exchange()
+          .expectStatus()
+          .isOk()
+          .expectHeader()
+          .contentType(expectedContentType)
+          .expectHeader()
+          .contentDisposition(
+              ContentDisposition.builder("attachment").filename(expectedFilename).build())
+          .expectBody(String.class)
+          .isEqualTo(expectedContent);
+    } catch (AssertionError e) {
+      return false;
```

Comment:
Idk how I feel about catching this assertion error exception, but it was the easiest way to deal with awaitility and still use assertions on the WebTestClient